source/FreeRTOS_DNS_Parser.c

```diff
@@ -48,6 +48,8 @@
 #if ( ipconfigUSE_DNS != 0 )
 
 
+    #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
```

Comment:
I'm OK with this PR, thank you.

Maybe you want to use the new access function `ipconfigIS_ENABLED()`:

~~~diff
     #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
     #if ( ipconfigIS_ENABLED( ipconfigUSE_DNS_CACHE ) ) ||
         ( ipconfigIS_ENABLED( ipconfigDNS_USE_CALLBACKS ) )
~~~

Reply:
Can we also enable this function for MDNS/LLMNR for [these lines](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/FreeRTOS_DNS_Parser.c#L467-L471)?
And also update [DNS_ParseDNSReply:358](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/FreeRTOS_DNS_Parser.c#L358) to handle DNS questions.

Otherwise, the compile would be failed when cache/callback/MDNS are disabled but LLMNR is enabled.

Reply:
So it would be like.

For DNS_ReadNameField:
~~~diff
-     #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
+     #if ( ipconfigIS_ENABLED( ipconfigUSE_DNS_CACHE ) ||
+           ipconfigIS_ENABLED( ipconfigDNS_USE_CALLBACKS ) ||
+           ipconfigIS_ENABLED( ipconfigUSE_MDNS ) ||
+           ipconfigIS_ENABLED( ipconfigUSE_LLMNR ) )
~~~

For DNS_ParseDNSReply:
~~~diff
-                    #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
+                    #if ( ipconfigIS_ENABLED( ipconfigUSE_DNS_CACHE ) ||
+                          ipconfigIS_ENABLED( ipconfigDNS_USE_CALLBACKS ) ||
+                          ipconfigIS_ENABLED( ipconfigUSE_MDNS ) ||
+                          ipconfigIS_ENABLED( ipconfigUSE_LLMNR ) )
                           if( x == 0U )
                           {
                               uxResult = DNS_ReadNameField( &xSet,
                                                             sizeof( xSet.pcName ) );
                               ( void ) uxResult;
                           }
                           else
                     #endif /* ipconfigUSE_DNS_CACHE || ipconfigDNS_USE_CALLBACKS || ipconfigUSE_MDNS || ipconfigUSE_LLMNR */
~~~

For [FreeRTOS_DNS_Globals.h](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/include/FreeRTOS_DNS_Globals.h#L192-L196):
~~~diff
-        #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 ) || ( ipconfigUSE_MDNS == 1 )
+        #if ( ipconfigIS_ENABLED( ipconfigUSE_DNS_CACHE ) ||
+              ipconfigIS_ENABLED( ipconfigDNS_USE_CALLBACKS ) ||
+              ipconfigIS_ENABLED( ipconfigUSE_MDNS ) ||
+              ipconfigIS_ENABLED( ipconfigUSE_LLMNR ) )
            BaseType_t xDoStore;                          /**< Becomes true when a DNS reply was requested by this device,
                                                           *   i.e. it has a matching request ID. */
            char pcName[ ipconfigDNS_CACHE_NAME_LENGTH ]; /**< A copy of the name that is mentioned in the questions. */
        #endif
~~~

Thank you.